Clients/Xamarin.Interactive.Client.Web/ClientApp/components/WorkbookShell.tsx

```diff
@@ -138,26 +140,61 @@ export class WorkbookShell extends React.Component<any, WorkbookShellState> {
         this.fileButton.click();
     }
 
-    loadWorkbook(event: React.ChangeEvent<HTMLInputElement>) {
+    async triggerGistPicker() {
+        if (!this.workbookEditor)
+            return;
+
+        const gistUrl = prompt("Enter the URL of the Gist you want to load from?");
+        if (!gistUrl)
+            return;
+
+        this.workbook = await loadWorkbookFromGist(this.shellContext.session, gistUrl);
+        await this.workbookEditor.loadNewContent(this.workbook.markdownContent)
+        await this.restoreNuGetPackages()
+    }
+
+    workbookFileChosen(event: React.ChangeEvent<HTMLInputElement>) {
         if (event.target.files == null) {
             alert("No files.");
             return;
         }
 
         const file = event.target.files[0]
         const reader = new FileReader
-        reader.addEventListener('load', () => this.restorePackages (reader))
-        reader.readAsText(file)
+        reader.addEventListener('load', () => this.loadWorkbook(file, reader))
+        reader.readAsArrayBuffer(file)
+    }
+
+    sniffHeader(file: ArrayBuffer): Boolean {
+        const magic = new Uint32Array(file.slice(0, 4))[0]
```

Comment:
Bounds check? What if 3 byte file?

Reply:
👍 I had tested this and Uint32Array didn't seem to care, but I tested wrong, it does.